### PR TITLE
Update allowed_hosts

### LINKED ISSUE #1076: Set Allowed Hosts for Site
- Currently in mysite/settings allowed sites is set to *, which is a potential security risk because it would allow anyone to host our site - should change to na-staging.herokuapp and production/newamerica.org

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/113920320650537/152981350214784)

### PR DESCRIPTION
**Pull request by jckpn**
Addresses #1076

### CODE REVIEW

File: newamericadotorg/settings/production.py [hunk 1]
```diff
@@ -15,8 +15,7 @@
 
 #ADMINS = [('Kirk', 'jackson@newamerica.org'), ('Andrew', 'lomax@newamerica.org')]
 
-# Will be changed to final host url
-ALLOWED_HOSTS = ['*']
+ALLOWED_HOSTS = ['www.newamerica.org']
```
Comment (nmorduch) [inline newamericadotorg/settings/production.py]: In order for `staging.newamerica.org` also to work, do we want to use `*.newamerica.org` here?

Comment (nmorduch) [reply newamericadotorg/settings/production.py]: Looks like my timing was off here!

Comment (jckpn) [reply newamericadotorg/settings/production.py]: Good point, *.newamerica.org would probably be better than www.newamerica.org.

Comment (jckpn) [reply newamericadotorg/settings/production.py]: Deleting the line in the dev settings may throw up errors as I think that's where `allowed_hosts` is defined, but I'll go ahead and make the array empty as the docs say, just to be safe.

**Comment by jckpn** [comment]
Deleting the line in the dev settings may throw up errors as I think that's where allowed_hosts is defined, but I'll go ahead and make the array empty as the docs say, just to be safe.

File: newamericadotorg/settings/production.py [hunk 2]
```diff
@@ -15,8 +15,10 @@
 
 #ADMINS = [('Kirk', 'jackson@newamerica.org'), ('Andrew', 'lomax@newamerica.org')]
 
-# Will be changed to final host url
-ALLOWED_HOSTS = ['*']
+ALLOWED_HOSTS = [
+    '.newamerica.org',
```
Comment (nmorduch) [inline newamericadotorg/settings/production.py]: Will this work? Should it not be `*.newamerica.org`?

Comment (jckpn) [reply newamericadotorg/settings/production.py]: Django uses a bit of a weird syntax for url wildcards. The docs says:

```
A value beginning with a period can be used as a subdomain wildcard: '.example.com' will match example.com, www.example.com, and any other subdomain of example.com.
```

Comment (nmorduch) [reply newamericadotorg/settings/production.py]: Oh weird. Thanks!

**Comment by nmorduch** [comment]
Local dev still works for me, not sure if there's a good way to test what this will do in production, and not confident that I have the expertise to assess the change. @kaedroho can you take a look?

Additionally, it would be great to squash the commits before merging this one.

File: newamericadotorg/settings/production.py (hunk 2 above)
Comment (kaedroho) [reply newamericadotorg/settings/production.py]: Nice, I didn't know this!

**Comment by nmorduch** [comment]
Assigning to myself to update the branch and merge.

**Comment by jckpn** [comment]
Sorry @nmorduch, I couldn't find a way to squash the commits without creating a new branch/pr altogether which I thought might mess with things -- just let me know if you still want me to :)

**Comment by nmorduch** [comment]
Noting here that I directed Jack to https://github.com/k88hudson/git-flight-rules#i-need-to-combine-commits

**Pull request merged**